lib/libzfs/libzfs_util.c

```diff
@@ -1060,6 +1060,7 @@ libzfs_init(void)
 		if ((error = zfs_nicestrtonum(hdl, env,
 		    &hdl->libzfs_max_nvlist))) {
 			errno = error;
+			free(hdl);
```

Comment:
It looks like we need to close `hdl->libzfs_fd` and `hdl->libzfs_mnttab` here too.

Reply:
@allanjude it'd be nice to wrap this one up.